### PR TITLE
SC-1595

### PR DESCRIPTION
**Pull request by anmol2302**
**Ticket Ref:** [SC-1595](https://project-sunbird.atlassian.net/browse/SC-1595)
**Release:** 2.6.5 
 Following API docs added
  - /v1/user/exists/{idType}/{id} (to check user exists or not)
  - /v1/user/signup 
  - /v1/user/get/{idType}/{id} to get user details

### CODE REVIEW

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://551.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://551.qa.docs.sunbird.org/2.6.0/apis/userapi// <br />  <br/> **Modified sections are -** <br />

File: apis/userapi/userapi.yaml [hunk 1]
```diff
@@ -168,6 +168,81 @@ paths:
         AllowDynamicQueryParameters: false
         AllowDynamicFormParameters: false
         IsMultiContentStreaming: false
+  /user/v3/create:
+    post:
+      description: >-
+        This is an Enhanced API with good throughput, is used to create a user
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: This api is lightweight of v2. This api will create user with very minimum attribute (firstName, email/phone, password, emailVerified/phoneVerified). All user created by this endpoint will be linked with custodian org.

File: apis/userapi/userapi.yaml [hunk 2]
```diff
@@ -168,6 +168,81 @@ paths:
...
+           - The endpoint for **Create User** is `/user/v2/create`
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: /user/v3/create

File: apis/userapi/userapi.yaml [hunk 3]
```diff
@@ -1721,6 +1796,135 @@ paths:
             'INTERNAL SERVER ERROR. We track these errors automatically and try to set it right at the earliest. Try refreshing the page. If the problem persists contact us at info@sunbird.org.'
           schema:
             $ref: '#/definitions/UserErrorResponse'
+  /user/v1/get/{idType}/{id}:
+    get:
+      description: >-
+        This API is used for searching a specific user, the user search is based on the unique ID type of the user
+          - The endpoint for **Get User by User ID** is `/user/v1/get/{idType}/{id}`
+          - The fields marked with an asterisk (*) are mandatory. They cannot be null or empty.
+          - User can pass an optional query parameter 'fields'. The value is a comma separated list of fields whose information is required in response. e.g. fields=completeness,missingFields,topic,organisations,roles
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: This line is not required for get user by email/phone
remove below as well.

File: apis/userapi/userapi.yaml [hunk 4]
```diff
@@ -1721,6 +1796,135 @@ paths:
...
+          in: header
+          required: false
+          type: string
+          description: 'This ID uniquely identifies a request if the same API is executed multiple times.'
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          description: 'Specify authorization key (format: Bearer api-key) received from administrator when performing given API request.'
+        - name: X-Authenticated-User-Token
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: - name: X-Authenticated-User-Token
 not required for get user by email/phone.

File: apis/userapi/userapi.yaml [hunk 5]
```diff
@@ -1721,6 +1796,135 @@ paths:
...
+            $ref: '#/definitions/UserErrorResponse'
+      security:
+        - bearer: []
+  /user/v1/exists/{idType}/{id}:
+    get:
+      description: >-
+        This API is used for searching a specific user, the user search is based on the unique ID type of the user
+          - The endpoint for **Get User by User ID** is `/user/v1/exists/{idType}/{id}`
+          - The fields marked with an asterisk (*) are mandatory. They cannot be null or empty.
+          - 'organisations' is to return organisation and location details.
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: organisations, roles not required.

File: apis/userapi/userapi.yaml [hunk 6]
```diff
@@ -1721,6 +1796,135 @@ paths:
...
+          required: false
+          type: string
+          description: 'This ID uniquely identifies a request if the same API is executed multiple times.'
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          description: 'Specify authorization key (format: Bearer api-key) received from administrator when performing given API request.'
+      responses:
+        '200':
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: Add response for 400 bad request as well.

File: apis/userapi/userapi.yaml [hunk 7]
```diff
@@ -168,6 +168,81 @@ paths:
...
+          in: body
+          required: true
+          description: >-
+            - The body payload comprises of the metadata of the user being created
+            - The body parameter refers to the format of the request, the request body (payload) takes the following media types to make the payload:
+
+               - JSON
+
+          schema:
+            $ref: '#/definitions/CreateUserRequestV2'
```
Comment (manzarul) [inline apis/userapi/userapi.yaml]: CreateUserRequestV2 create CreateUserRequestV3 which will support following attributes only.
 firstName, email, phone, emailVerified, phoneVerified,password

File: apis/userapi/userapi.yaml (hunk 2 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

File: apis/userapi/userapi.yaml (hunk 1 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

File: apis/userapi/userapi.yaml (hunk 3 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

File: apis/userapi/userapi.yaml (hunk 4 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: its required @manzarul

File: apis/userapi/userapi.yaml (hunk 5 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

File: apis/userapi/userapi.yaml (hunk 6 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

File: apis/userapi/userapi.yaml (hunk 7 above)
Comment (anmol2302) [reply apis/userapi/userapi.yaml]: done

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://551.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://551.qa.docs.sunbird.org/2.6.0/apis/userapi// <br />  <br/> **Modified sections are -** <br />

File: apis/userapi/userapi.yaml (hunk 4 above)
Comment (manzarul) [reply apis/userapi/userapi.yaml]: ok

**Comment by manzarul** [comment]
Since i won't have option to resolved comment, it's verified now and can be merged with relase-2.5.6

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://551.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://551.qa.docs.sunbird.org/2.6.0/apis/userapi// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://551.qa.docs.sunbird.org/2.6.0/ .<br/> **Note:** Only _2.6.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://551.qa.docs.sunbird.org/2.6.0/apis/userapi// <br />  <br/> **Modified sections are -** <br />

**Pull request merged**